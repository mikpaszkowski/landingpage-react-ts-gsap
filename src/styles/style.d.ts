import "styled-components";

interface IPalette {
  footerBackground: string;
  font: string;
  fontLight: string;
  playButton: string;
}

declare module "styled-components" {
  export interface DefaultTheme {
    palette: {
      common: {
        black: string;
        white: string;
        theme: string;
        background: string;
      };
      primary: IPalette;
      secondary: IPalette;
    };
  }
}
