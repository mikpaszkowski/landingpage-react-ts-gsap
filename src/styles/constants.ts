interface Fonts {
  primaryFont: string;
  secondaryFont: string;
}

interface BreakPoints {
  mobileSmall: string;
  mobileMedium: string;
  mobileLarge: string;
  tablet: string;
  tabletLarge: string;
  laptopSmall: string;
  laptopLarge: string;
}

export const fontTypes: Fonts = {
  primaryFont: "'Rozha One', serif",
  secondaryFont: "'Reem Kufi', sans-serif",
};

export const breakPoints: BreakPoints = {
  mobileSmall: "320px",
  mobileMedium: "375px",
  mobileLarge: "525px",
  tablet: "768px",
  tabletLarge: "1110px",
  laptopSmall: "1280px",
  laptopLarge: "1440px",
};

export {};
