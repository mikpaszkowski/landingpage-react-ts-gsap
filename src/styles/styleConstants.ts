interface Fonts {
  primaryFont: string;
  secondaryFont: string;
}

interface BreakPoints {
  mobileMedium: string;
  laptopSmall: string;
  laptopLarge: string;
}

export const fontTypes: Fonts = {
  primaryFont: "'Rozha One', serif",
  secondaryFont: "'Reem Kufi', sans-serif",
};

export const breakPoints: BreakPoints = {
  mobileMedium: "375px",
  laptopSmall: "1200px",
  laptopLarge: "1440px",
};

export {};
