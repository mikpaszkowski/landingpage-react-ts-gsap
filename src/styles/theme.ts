import { DefaultTheme } from "styled-components";

export const lightTheme: DefaultTheme = {
  palette: {
    common: {
      black: "#131E07",
      white: "#FFFFFF",
      theme: "#FABD20",
      background: "#FEF4E6",
    },
    primary: {
      footerBackground: "#131E07",
      font: "#131E07",
      fontLight: "#E5E5E5",
      playButton: "#131E07",
    },
    secondary: {
      footerBackground: "#131E07",
      font: "#40462F",
      fontLight: "#7C8176",
      playButton: "#131E07",
    },
  },
};

export const darkTheme: DefaultTheme = {
  palette: {
    common: {
      black: "#131E07",
      white: "#FFFFFF",
      theme: "#FABD20",
      background: "#131E07",
    },
    primary: {
      footerBackground: "#131E07",
      font: "#FFFFFF",
      fontLight: "#E5E5E5",
      playButton: "#131E07",
    },
    secondary: {
      footerBackground: "#131E07",
      font: "#E5E5E5",
      fontLight: "#7C8176",
      playButton: "#131E07",
    },
  },
};
