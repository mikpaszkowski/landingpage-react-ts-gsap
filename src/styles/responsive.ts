import { breakPoints } from "./styleConstants";

interface DeviceResponsive {
  mobileMedium: string;
  laptopSmall: string;
  laptopLarge: string;
}

export const device: DeviceResponsive = {
  mobileMedium: `(max-width: ${breakPoints.mobileMedium})`,
  laptopSmall: `(max-width: ${breakPoints.laptopSmall})`,
  laptopLarge: `(max-width: ${breakPoints.laptopLarge})`,
};
