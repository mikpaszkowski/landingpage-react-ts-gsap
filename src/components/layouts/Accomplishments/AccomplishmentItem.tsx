import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../../styles/constants";
import { device } from "../../../styles/responsive";

const AccomplishmentItemWrapper = styled.div``;

const Headline = styled.h1`
  display: block;
  font-family: ${fontTypes.primaryFont};
  font-style: normal;
  font-weight: normal;
  font-size: 8rem;
  line-height: 117%;
  margin-bottom: 1.6rem;
  text-align: center;
  font-feature-settings: "haln" on;
  color: ${({ theme }) => theme.palette.common.theme};

  @media ${device.mobileMedium} {
    margin-bottom: 0.6rem;
  }
`;

const Description = styled.p`
  display: block;
  font-family: ${fontTypes.secondaryFont};
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
  margin: 0 4rem;
  text-align: center;
  color: ${({ theme }) => theme.palette.primary.font};
`;

interface AccomplishmentProps {
  headline: string;
  description: string;
}

export const AccomplishmentItem = ({
  headline,
  description,
}: AccomplishmentProps) => {
  return (
    <AccomplishmentItemWrapper>
      <Headline>{headline}</Headline>
      <Description>{description}</Description>
    </AccomplishmentItemWrapper>
  );
};
