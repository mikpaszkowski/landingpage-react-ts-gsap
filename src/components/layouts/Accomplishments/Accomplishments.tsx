import * as React from "react";

import styled from "styled-components";
import { AccomplishmentItem } from "./AccomplishmentItem";
import { accomplishmentItems } from "../../../assets/accomplishmentsContent/accomplishmentItemList";
import { device } from "../../../styles/responsive";
import gsap from "gsap";
import { ScrollTrigger } from "gsap/dist/ScrollTrigger";
import { laptopLarge, mobileMedium } from "../../../animations/accomplishments";

gsap.registerPlugin(ScrollTrigger);

const AccomplishmentsWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 11.1rem;
  margin: 22rem 21.9rem;
  will-change: transform;

  @media ${device.mobileMedium} {
    display: block;
    margin: 6.4rem 0;
    padding: 0 2.4rem;
  }
`;

export const Accomplishments = () => {
  const { useRef, useEffect } = React;

  const sectionRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    ScrollTrigger.matchMedia({
      "(min-width: 1440px)": () => laptopLarge(sectionRef),

      "(max-width: 375px)": () => mobileMedium(sectionRef),
    });
  });

  return (
    <AccomplishmentsWrapper ref={sectionRef}>
      {accomplishmentItems.map(({ id, ...props }) => (
        <AccomplishmentItem key={id} {...props} />
      ))}
    </AccomplishmentsWrapper>
  );
};
