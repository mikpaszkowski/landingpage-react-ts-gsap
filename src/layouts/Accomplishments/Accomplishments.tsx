import * as React from "react";

import styled from "styled-components";
import { AccomplishmentItem } from "./AccomplishmentItem";
import { accomplishmentItems } from "../../data/staticContents/accomplishments/accomplishmentItemList";
import { device } from "../../styles/responsive";
import gsap from "gsap";
import { ScrollTrigger } from "gsap/dist/ScrollTrigger";
import {
  laptopLarge,
  mobileMedium,
  tabletLarge,
} from "../../animations/accomplishments";

gsap.registerPlugin(ScrollTrigger);

const AccomplishmentsWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 11.1rem;
  margin: 22rem 21.9rem;
  will-change: transform;

  @media ${device.laptopSmall} {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: center;
    align-content: center;
    margin: 22rem 21.9rem;
  }

  @media ${device.tabletLarge} {
    margin: 15rem 0;
    flex-direction: column;
  }

  @media ${device.mobileLarge} {
    margin: 6.4rem 0;
    flex-direction: column;
  }
`;

export const Accomplishments = () => {
  const { useRef, useEffect } = React;

  const sectionRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    ScrollTrigger.matchMedia({
      "(min-width: 1440px)": () => laptopLarge(sectionRef),
      "(min-width: 500px)": () => tabletLarge(sectionRef),
      "(max-width: 499px)": () => mobileMedium(sectionRef),
    });
  });

  return (
    <AccomplishmentsWrapper ref={sectionRef}>
      {accomplishmentItems.map(({ id, ...props }) => (
        <AccomplishmentItem key={id} {...props} />
      ))}
    </AccomplishmentsWrapper>
  );
};
