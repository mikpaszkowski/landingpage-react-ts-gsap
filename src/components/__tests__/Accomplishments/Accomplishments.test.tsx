import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import { Accomplishments } from "../../layouts/Accomplishments/Accomplishments";
import { ThemeProvider } from "styled-components";
import { lightTheme } from "../../../styles/theme";

afterEach(() => {
  cleanup();
});

it("check render correctly <Accomplishments /> component", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={lightTheme}>
        <Accomplishments />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
