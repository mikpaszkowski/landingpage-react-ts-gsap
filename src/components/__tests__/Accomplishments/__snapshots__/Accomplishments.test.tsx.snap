// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render correctly <Accomplishments /> component 1`] = `
.c2 {
  display: block;
  font-family: 'Rozha One',serif;
  font-style: normal;
  font-weight: normal;
  font-size: 8rem;
  line-height: 117%;
  margin-bottom: 1.6rem;
  text-align: center;
  font-feature-settings: "haln" on;
  color: #FABD20;
}

.c3 {
  display: block;
  font-family: 'Reem Kufi',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
  margin: 0 4rem;
  text-align: center;
  color: #131E07;
}

.c0 {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 11.1rem;
  margin: 22rem 21.9rem;
  will-change: transform;
}

@media (max-width:375px) {
  .c1 {
    margin-bottom: 10rem;
  }
}

@media (max-width:375px) {
  .c2 {
    margin-bottom: 0.6rem;
  }
}

@media (max-width:375px) {
  .c0 {
    display: block;
    margin: 6.4rem 0;
    padding: 0 2.4rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      230+
    </h1>
    <p
      className="c3"
    >
      Scientifically based tests
    </p>
  </div>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      10,000+
    </h1>
    <p
      className="c3"
    >
      Medical recommendations
    </p>
  </div>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      25M+
    </h1>
    <p
      className="c3"
    >
      Students learning
    </p>
  </div>
</div>
`;
